models/gif_psc_exp_multisynapse.cpp

```diff
@@ -0,0 +1,492 @@
+
+  ArrayDatum tau_syn_ad( tau_syn_ );
+  def< ArrayDatum >( d, names::taus_syn, tau_syn_ad );
+
+  ArrayDatum tau_sfa_list_ad( tau_sfa_ );
+  def< ArrayDatum >( d, names::tau_sfa, tau_sfa_list_ad );
+
+  ArrayDatum q_sfa_list_ad( q_sfa_ );
+  def< ArrayDatum >( d, names::q_sfa, q_sfa_list_ad );
+
+  ArrayDatum tau_stc_list_ad( tau_stc_ );
+  def< ArrayDatum >( d, names::tau_stc, tau_stc_list_ad );
+
+  ArrayDatum q_stc_list_ad( q_stc_ );
+  def< ArrayDatum >( d, names::q_stc, q_stc_list_ad );
+}
+
+void
+nest::gif_psc_exp_multisynapse::Parameters_::set( const DictionaryDatum& d )
+{
+  updateValue< double >( d, names::I_e, I_e_ );
+  updateValue< double >( d, names::E_L, E_L_ );
+  updateValue< double >( d, names::g_L, g_L_ );
+  updateValue< double >( d, names::C_m, c_m_ );
+  updateValue< double >( d, names::V_reset, V_reset_ );
+  updateValue< double >( d, names::Delta_V, Delta_V_ );
+  updateValue< double >( d, names::V_T_star, V_T_star_ );
+
+  if ( updateValue< double >( d, names::lambda_0, lambda_0_ ) )
+  {
+    lambda_0_ /= 1000.0; // convert to 1/ms
+  }
+
+  updateValue< double >( d, names::t_ref, t_ref_ );
+
+  updateValue< std::vector< double > >( d, names::tau_sfa, tau_sfa_ );
+  updateValue< std::vector< double > >( d, names::q_sfa, q_sfa_ );
+  updateValue< std::vector< double > >( d, names::tau_stc, tau_stc_ );
+  updateValue< std::vector< double > >( d, names::q_stc, q_stc_ );
+
+  if ( tau_sfa_.size() != q_sfa_.size() )
+    throw BadProperty( String::compose(
+      "'tau_sfa' and 'q_sfa' need to have the same dimensions.\nSize of "
+      "tau_sfa: %1\nSize of q_sfa: %2",
+      tau_sfa_.size(),
+      q_sfa_.size() ) );
+
+  if ( tau_stc_.size() != q_stc_.size() )
+    throw BadProperty( String::compose(
+      "'tau_stc' and 'q_stc' need to have the same dimensions.\nSize of "
+      "tau_stc: %1\nSize of q_stc: %2",
+      tau_stc_.size(),
+      q_stc_.size() ) );
+
+  if ( g_L_ <= 0 )
+    throw BadProperty( "Membrane conductance must be strictly positive." );
+
+  if ( Delta_V_ <= 0 )
+    throw BadProperty( "Delta_V must be strictly positive." );
+
+  if ( c_m_ <= 0 )
+    throw BadProperty( "Capacitance must be strictly positive." );
+
+  if ( t_ref_ < 0 )
+    throw BadProperty( "Refractory time must not be negative." );
+
+  if ( lambda_0_ < 0 )
+    throw BadProperty( "lambda_0 must not be negative." );
+
+  for ( size_t i = 0; i < tau_sfa_.size(); i++ )
+    if ( tau_sfa_[ i ] <= 0 )
+      throw BadProperty( "All time constants must be strictly positive." );
+
+  for ( size_t i = 0; i < tau_stc_.size(); i++ )
+    if ( tau_stc_[ i ] <= 0 )
+      throw BadProperty( "All time constants must be strictly positive." );
+
+  std::vector< double > tau_tmp;
+  if ( updateValue< std::vector< double > >( d, names::taus_syn, tau_tmp ) )
+  {
+    for ( size_t i = 0; i < tau_tmp.size(); ++i )
+    {
+      if ( tau_tmp.size() < tau_syn_.size() && has_connections_ == true )
```

Comment:
This test is for the entire tau-vector, so it should be outside the loop. Actually, inside the loop it would not detect a problem if the user passed an empty array of time constants, because the loop would not be executed at all.

Also, avoid comparisons with truth values. I think the test would be more expressive with the `has_connections_` first:
```C++
if ( has_connections_ and tau_tmp.size() < tau_syn_.size() )
{
  throw ...
}
for ( ...
```